examples/ensemble_examples.py

```diff
@@ -0,0 +1,132 @@
+# 
+# Most parameterizations and initalizations therein may not result in optimal performance or detection, and are provided just for demonstration.
+# 
+# ## Imports
+
+# In[ ]:
+
+
+import numpy as np
+
+from menelaus.concept_drift import STEPD
+from menelaus.datasets import make_example_batch_data, fetch_rainfall_data
+from menelaus.data_drift import HDDDM, KdqTreeBatch, KdqTreeStreaming
+from menelaus.ensemble import BatchEnsemble, StreamingEnsemble
+from menelaus.ensemble import SimpleMajorityElection, MinimumApprovalElection
+
+
+# ## Import Data
+
+# In[ ]:
+
+
+example_data = make_example_batch_data()
+rainfall_data = fetch_rainfall_data()
+
+
+# ## Batch Ensemble
+# 
+# The simplest use of an ensemble is to combine three data-drift-only detectors with few additional settings. In this case we can combine three instances of batch detectors (`KdqTreeBatch`, `HDDDM`), all operating on the same data columns, with a very basic evaluation scheme (*i.e.* a simple majority of detectors alarming, causes the ensemble to alarm).
+
+# In[ ]:
+
+
+# initialize set of detectors with desired parameterizations
+detectors = {
+    'k1': KdqTreeBatch(bootstrap_samples=500),
+    'k2': KdqTreeBatch(bootstrap_samples=475),
+    'h1': HDDDM()
+}
+
+# choose an election scheme
+# SimpleMajorityElection will alarm when more than half of the detectors are in 
+# the "drift" state.
+election = SimpleMajorityElection()
+
+# initialize an ensemble object
+ensemble = BatchEnsemble(detectors, election)
+
+
+# Note that `BatchEnsemble` and `StreamingEnsemble` are instances of `BatchDetector` and `StreamingDetector` themselves (respectively). As such, they are used in the same way syntactically and possess similar properties.
+
+# In[ ]:
+
+
+# make dataset smaller
+df_example_data = example_data[example_data.year < 2010]
+
+# split dataset into 1 dataset for each 'batch' (year)
+df_into_batches = [x for _,x in df_example_data.groupby('year')]
+df_into_batches = [x[['a', 'b', 'c']] for x in df_into_batches]
+
+# batch detectors -- and ensembles -- need an initial reference batch
+ensemble.set_reference(df_into_batches[0])
+print(f"Batch #{0} | Ensemble reference set")
+
+for i, batch in enumerate(df_into_batches[1:]):
+    ensemble.update(batch)
+    print(f"Batch #{i+1} | Ensemble overall drift state: {ensemble.drift_state}")
+
+
+# ## Streaming Ensemble
+# 
+# Using an ensemble of streaming detectors can involve additional features. This example uses both data and concept drift detectors (`KdqTreeStreaming`, `STEPD`), custom subsets of data for different detectors, as well as a different election scheme that can mitigate internal differences in how different detectors 'wait' before alarming for drift.
```

Comment:
The "different election scheme" here sounds like it's referring to ConfirmedApproval. Should the example or the narrative be modified (in the notebook)?